new-chatbot-app/src/llm-chain/llm-toolbox/libcal-tools/cancel-reservation/cancel-reservation.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import { NetworkService } from 'src/shared/services/network/network.service';
+import { Room, LibcalApiAuthorizationService } from '../../../../library-api/libcal-api-auth/libcal-api-auth.service';
+
+@Injectable()
+export class CancelReservationService {
+
+  public readonly name: string = "CancelReservationService";
+  public readonly description: string =
+    "This tool is for cancelling study room reservation. Use Final Answer instead if you don't have enough required parameters yet. Don't include any single quotes in the paramter.";
+
+  public readonly parameters: { [parameterName: string]: string } = {
+    bookingID: "string [REQUIRED]"
+  };
+
+  /**
+   * Singleton instance of the CancelReservationService.
+   */
+  constructor(
+    private networkService: NetworkService,
+    private libcalApiAuthService: LibcalApiAuthorizationService
+  ) {}
+
+  /**
+   * Run the cancel reservation service.
+   * @param toolInput 
+   * @returns
+   */
+  async serviceRun(toolInput: { bookingID: string | null }): Promise<string> {
+    return new Promise<string>(async (resolve, reject) => {
```

Comment:
new version of Typescript doesn't require this anymore. I suggest remove this and simply return instead of resolve